Sources/EasyPurchase/PaymentsController.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  PaymentsController.swift
+//  
+//
+//  Created by BJIT on 22/9/23.
+//
+
+import Foundation
+import StoreKit
+
+// Structure representing a payment
+struct Payment {
+    let product: SKProduct         // The product being purchased
+    let quantity: Int              // The quantity of the product (e.g., for consumable items)
+    var needToDownloadContent: Bool // Indicates whether content needs to be downloaded after purchase
+    var completion: (PurchaseResult) -> Void // Completion block to handle purchase result
+}
+
+// Enum representing the result of a purchase
+enum PurchaseResult {
+    case success
+    case failure(error: Error?)
+}
+
+// Protocol defining transaction handling methods
+protocol TransactionController {
+    func processTransactions(_ transactions: [SKPaymentTransaction], on paymentQueue: PaymentQueue) -> [SKPaymentTransaction]
+}
+
+// Implementation of the transaction controller
+class PaymentsController: TransactionController {
+
+    var payments: [Payment] = []          // Array to hold pending payments
+    var unhandledTransactions: [SKPaymentTransaction] = [] // Array to hold unhandled transactions
```

Comment:
var unhandledTransactions: [SKPaymentTransaction] = []

-> Why store this property as we do not need it and it's a stateless.

Reply:
Noted